src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java

```diff
@@ -74,6 +76,20 @@
             utilityFunctions)
             .flatMap(Set::stream).collect(Collectors.toSet());
 
+    private HashMap<String, Integer> generatedIds = new HashMap<>();
+
+    /**
+     * Generates next id for given method name. The id's are increasing for each method name, so
+     * nextId("a"), nextId("a"), nextId("b") will return a_1, a_2, b_1
+     */
+    public String nextId(String methodName) {
```

Comment:
Not sure if this look clear to you but it seems work for your purpose:

```
  private String nextId(String methodName) {
    return methodName + generatedIds.merge(methodName, 1, Integer::sum);
  }
```

Reply:
Nice, i like it, thanks, will do